### PR TITLE
chore: typing updates

### PR DESCRIPTION
**Pull request by henryiii**
Some minor cleanup, and updates to typing.

- chore: remove wheel
- chore: minor style issue with pytest config
- typing: update mypy typing, fix Iterable -> Collection

### CODE REVIEW

**Review by layday — APPROVED** [review]

File: src/build/env.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def executable(self) -> str:
     def scripts_dir(self) -> str:
         return self._scripts_dir
 
-    def install(self, requirements: Iterable[str]) -> None:
+    def install(self, requirements: Collection[str]) -> None:
```
Comment (henryiii) [inline src/build/env.py:186]: FYI, this is checked just below (`if not requirements`), which is invalid on an `Iterable` - it just happens to be true for some common iterables like set, list, and tuple. Caught by `"truthy-bool"`. I've changed it to `Collection`.

File: setup.cfg [hunk 2]
```diff
@@ -63,7 +63,7 @@ test =
     setuptools>=56.0.0;python_version >= "3.10"
 typing =
     importlib-metadata>=4.6.4
-    mypy==0.931
+    mypy==0.941
```
Comment (gaborbernat) [inline setup.cfg:66]: Any reason why not use `0.942`?

Comment (henryiii) [reply setup.cfg:66]: Oops, forgot! No.

```suggestion
    mypy==0.942
```

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**